forum/notifications/models.py

```diff
@@ -1,3 +1,18 @@
 from django.db import models
+from django.contrib.auth.models import User
 
-# Create your models here.
+
+class NotificationType(models.Model):
+    name = models.CharField(max_length=100, unique=True)
+    description = models.TextField()
+
+    def __str__(self):
+        return self.name
+
```

Comment:
The __str__ method for NotificationPreference can check for the existence of self.user.username to avoid errors if the user has been deleted:
def __str__(self):
    return f"Preferences for {self.user.username if self.user else 'Unknown User'}"